Formula/gemstash.rb

```diff
@@ -0,0 +1,124 @@
+class Gemstash < Formula
+  desc "RubyGems.org cache and private gem server"
+  homepage "https://github.com/bundler/gemstash"
+  url "https://rubygems.org/downloads/gemstash-2.0.0.gem"
+  sha256 "511a5773c43507b913c4f2f57d33a3d972bd71378b2669c22a357b3351a82bf4"
+  head "https://github.com/bundler/gemstash.git"
+
+  if MacOS.version < :mojave
+    depends_on "openssl"
+    depends_on "ruby"
+  end
+
+  def install
+    if build.head?
+      system "gem", "build", "gemstash.gemspec"
+      gem_file = Dir["*.gem"].first
+    else
+      gem_file = "gemstash-#{version}.gem"
+    end
+
+    ENV["GEM_HOME"] = libexec
+
+    if MacOS.version < :mojave
+      gem = Formula["ruby"].bin/"gem"
+    else
+      gem = "gem"
+    end
+
+    system gem, "install", "--no-document", gem_file
+
+    (bin/"gemstash").write_env_script libexec/"bin/gemstash",
+      :GEM_HOME                      => ENV["GEM_HOME"],
+      :SINATRA_ACTIVESUPPORT_WARNING => false
+
+    (var/"gemstash").mkpath
+    (etc/"gemstash").mkpath
+    config_file.write <<~EOS
```

Comment:
Config should be provided by upstream or the user

Reply:
I'm not sure how to proceed with the config file because the only reason it's picked up by the server is because I have configured the service to look for the file: https://github.com/Homebrew/homebrew-core/pull/39638/files#diff-84922145f9de0e0337eb37d83971815aR76

If I remove the config file I also have to remove the launch parameter, which means that if a user wants to use a config file at all they will have to edit the plist.